src/server/src/main/java/io/cassandrareaper/service/RepairRunService.java

```diff
@@ -185,7 +196,32 @@ public RepairRun registerRepairRun(
       LOG.error(errMsg);
       throw new ReaperException(errMsg);
     }
-    return segments;
+
+    // Compute replicas per DC for each segment
+    List<Segment> segmentsWithReplicas = Lists.newArrayList();
+    for (Segment segment:segments) {
+      segmentsWithReplicas.add(
+          Segment.builder()
+                 .withBaseRange(segment.getBaseRange())
+                 .withTokenRanges(segment.getTokenRanges())
+                 .withReplicas(getDCsByNodeForRepairSegment(targetCluster, segment, repairUnit.getKeyspaceName()))
+                 .build());
+    }
+
+    return segmentsWithReplicas;
+  }
+
+  private Map<String, String> getDCsByNodeForRepairSegment(
```

Comment:
my opinion is that this method should be in the `SegmentRunner` class.

and, the existing `getDCsByNodeForRepairSegment` there should be removed as no one is using it now.
https://github.com/thelastpickle/cassandra-reaper/blob/99ff8fc1743106cc3905a916ee44641d0a7ea6b0/src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java#L750

Reply:
I agree that the existing one in the `SegmentRunner` class should be removed, but I don't see a reason why the one from `RepairRunService` should move to `SegmentRunner`. It's used in `RepairRunService.generateSegments()` only and we don't have a SegmentRunner object instanciated at that stage.

Reply:
> I don't see a reason why the one from RepairRunService should move to SegmentRunner. It's used in RepairRunService.generateSegments() only and we don't have a SegmentRunner object instanciated at that stage.

Agreed.